service/config_provider.go

```diff
@@ -213,8 +205,8 @@ func (cm *configProvider) Shutdown(ctx context.Context) error {
 // https://tools.ietf.org/id/draft-kerwin-file-scheme-07.html#syntax
 var driverLetterRegexp = regexp.MustCompile("^[A-z]:")
 
-func (cm *configProvider) mergeRetrieve(ctx context.Context) (configmapprovider.Retrieved, error) {
-	var retrs []configmapprovider.Retrieved
+func (cm *configProvider) mergeRetrieve(ctx context.Context) (*configmapprovider.Retrieved, error) {
+	var retrs []configmapprovider.CloseFunc
```

Comment:
```suggestion
	var closers []configmapprovider.CloseFunc
```

This is now a different type of list and should be named accordingly.

Reply:
Done.